File: src/vovka_pkg/src/consumer.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"

void callback(const std_msgs::String::ConstPtr & msg) {

    ROS_INFO("WOW, I've just heard about %s, it's awesome!", msg->data.c_str());
}


int main(int argc, char **argv) {
    
    ros::init(argc, argv, "consumer");
    ros::NodeHandle node;
    ros::Subscriber subscriber = node.subscribe("market", 5, callback);
    ros::spin();

    return 0;
}

File: src/vovka_pkg/scripts/consumer.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import String


def callback(msg):
    rospy.loginfo(rospy.get_caller_id() + ' WOW! Have you heard about %s', msg.data)


def consumer():
    rospy.init_node('consumer', anonymous=True)
    rospy.Subscriber('market', String, callback)

    rospy.spin()


if __name__ == '__main__':
    consumer()
